Modules/_sqlite/statement.c

```diff
@@ -377,24 +374,15 @@ int pysqlite_statement_reset(pysqlite_Statement* self)
 
     rc = SQLITE_OK;
 
-    if (self->in_use && self->st) {
+    if (self->st && sqlite3_stmt_busy(self->st)) {
```

Comment:
`sqlite3_stmt_busy` also does a NULL pointer check, but we'll avoid a function call if we do it ourselves. Resetting a statement is a frequent operation.

Reply:
Aah, if I'm to add this optimisation back, I need to harden the exit points in the query execute code. But that's a nice side effect. I'll mark this as draft until I'm done.

Reply:
Actually, I think it's better to leave the optimisation out and rather optimise the API usage in a later PR; that is, only call reset when needed.